Src/Support/Google.Apis.Auth/OAuth2/Responses/TokenResponse.cs

```diff
@@ -110,28 +110,73 @@ public DateTime Issued
         [JsonProperty("expireTime")]
         private string ImpersonatedAccessTokenExpireTime { get; set; }
 
-        // Note: ideally this would be called ShouldRefresh or similar.
+        /// <summary>
+        /// Returns true if the token represented by this token response should be refreshed.
+        /// Note that this may be true for valid tokens, in which case a pre-emptive refresh is adviced
+        /// even if the current token may be used while it continues to be valid.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="MayBeUsed(IClock)"/> for information on when a token is considered valid.
+        /// A valid token is considered stale if it's close to expiring, but not so much as to be unusable.
+        /// </remarks>
+        [JsonIgnore]
+        public bool IsStale => ShouldBeRefreshed(SystemClock.Default);
+
+        /// <summary>
+        /// Returns true if the token represented by this token response is valid, that is, it may be used
+        /// for authentication and authorizations purposes.
+        /// </summary>
+        /// <remarks>
+        /// A token is considered valid if all of the following are true:
+        /// <list type="bullet">
+        /// <item>At least one of <see cref="AccessToken"/> and <see cref="IdToken"/> is not null.</item>
+        /// <item><see cref="ExpiresInSeconds"/> is not null.</item>
+        /// <item>The token has not expired and will not expire in the very near future. That is if
```

Comment:
Hmm. This is where I have a problem with "valid" as the term. If a token hasn't expired, I would say it's valid. If I send it to a service and there *isn't* clock skew involved, it's still going to succeed.

Do we need to make this public immediately? Can we make it internal for now (or even remove it if we're using a clock elsewhere) and then make it public when we've got a better name?